src/middlewared/middlewared/common/smart/smartctl.py

```diff
@@ -31,7 +32,15 @@ async def get_smartctl_args(context, disk, smartoptions):
         return [f"/dev/{disk}", "-d", "nvme"] + smartoptions
 
     args = [f"/dev/{disk}"] + smartoptions
-    if not enterprise_hardware and device['bus'] == 'USB':
+
+    sat = False
+    if enterprise_hardware:
+        if await context.middleware.run_in_thread(os.path.exists, f"/sys/block/{disk}/device/vpd_pg89"):
```

Comment:
Instead of passing around `middleware` object, you can do `asyncio.to_thread`. Just a thought.

Reply:
Using `asyncio.to_thread` will have to spawn a new thread for every disk, I like using our pool